src/Runner.Common/JobServer.cs

```diff
@@ -2,16 +2,19 @@
 using System;
 using System.Collections.Generic;
 using System.IO;
+using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
+using GitHub.Runner.Sdk;
 using GitHub.Services.WebApi;
+using GitHub.Services.Common;
 
 namespace GitHub.Runner.Common
 {
     [ServiceLocator(Default = typeof(JobServer))]
     public interface IJobServer : IRunnerService
     {
-        Task ConnectAsync(VssConnection jobConnection);
+        Task ConnectAsync(Uri jobServerUrl, VssCredentials jobServerCredential, IThrottlingReporter jobServerQueue = null);
```

Comment:
rename `jobServerQueue` to `throttlingReporter`

Reply:
wonder whether `ConnectAsync` should just take the optional `DelegatingHandler[]` instead of the `IThrottlingReporter`?